### DISCUSSION TITLE
Escaping of simple quotes in callback on the command line

### DISCUSSION THREAD

**Discussion by AlexWeinreb** [GMOD/jbrowse-components#3361] [Q&A] [answered]
[The documentation](https://jbrowse.org/jb2/docs/faq/#adding-color-callbacks-via-the-command-line) shows one way to add a callback from the command line:

```
jbrowse add-track somevariants.vcf --load copy --config '{"displays": [{"displayId": "my_BasicDisplay", "type": "LinearBasicDisplay", "renderer": {"color1": "jexl:get(feature, 'strand') == -1 ? 'red' : 'blue'" }}]}'
```
However if I try to use this, it fails (all annotations are black). This is apparently due to the simple quotes `'` being removed: opening that track's settings in the browser has "color1" set to
```
get(feature, strand) == -1 ? red : blue
```
and adding back the quotes displays the correct colors (similarly, config.json does not have the quotes, adding them manually yields correct coloring).

I tried escaping the single quotes with `\` or `\\`, but this gives a syntax error.

Jbrowse version:
@jbrowse/cli/2.2.1 linux-x64 node-v14.18.0

**Comment by AlexWeinreb** [comment, marked as answer]
Turns out a solution is:
```
--config '{"displays": [{"displayId": "my_BasicDisplay", "type": "LinearBasicDisplay", "renderer": {"color1": "jexl:get(feature, '\''strand'\'') == -1 ? '\''red'\'' : '\''blue'\''" }}]}'
```
From [here](https://riptutorial.com/bash/example/2465/quoting-literal-text), the 4-characters sequence `'\''` is equivalent to a literal `'`.

It might still be useful to update the docs, and I'd be interested if anyone knows of a more readable solution.

**Reply by cmdcolin** [reply]
nice catch, we will update docs with this!
